lib/timers.js

```diff
@@ -87,13 +87,20 @@ function listOnTimeout() {
 
   var first;
   while (first = L.peek(list)) {
-    // If the previous iteration caused a timer to be added,
-    // update the value of "now" so that timing computations are
-    // done correctly. See test/simple/test-timers-blocking-callback.js
-    // for more information.
+    // Whenever a timer creates a new timer in the current list being processed
```

Comment:
I think your commit message is fine, with the exception that it says that we process timers added within timers' callbacks immediately in _all cases_ when the two timers' timeouts are equal. Actually that happens _only_ if the time to run through all previously scheduled timers with the same timeout value exceeds the timers' timeout.

This is likely to happen with timers with short timeouts, like with `setTimeout(function() { setTimeout(function() {}, 0); }, 0);`. However it would not happen with longer ones, unless there's blocking code that takes a while to run in the first timer's callback.

So I would rephrase the beginning of your commit message as such:

```
Whenever a timer with a specific timeout value creates a new timer with the same
timeout, the newly added timer might be processed immediately in the same tick
of the event loop instead of during the next tick of the event loop at the earliest.
```